### ISSUE TITLE
Manyfold3D Marketplace App broken

### ISSUE THREAD

**Issue by zacgoodwin** [azukaar/cosmos-servapps-official#169]
Using default settings I get this error: 

2024-12-07 20:10:09
Preparing database...
2024-12-07 20:10:10
bin/rails aborted!
2024-12-07 20:10:10
URI::InvalidURIError: the scheme postgresql does not accept registry part:  (or bad hostname?) (URI::InvalidURIError)

I updated the DB URL from postgresql://:@/?pool=5 tp  to postgresql://Manyfold-postgres:5432/?pool=5 along with other strings I could think of and didn't have any luck. Does anyone else have ideas? 

Should be able to reproduce with a default cosmos install and a default manyfold install

**Comment by kolossboss** [comment]
I have the same issue.

Installing Manyfold via offical Cosmos Appstore results in an Error with the creating of the database:

` URI::InvalidURIError: the scheme postgresql does not accept registry part: :@ (or bad hostname?) (URI::InvalidURIError)`

I think the reasons is, the DB_USER ...variables are not added to the Base URL`"DATABASE_URL=postgresql://{DB_USER}:{DB_PASS}@{DB_HOST}/{DB_NAME}?pool=5"`

So I added them manually.

The DB is now created.

But I still cant receive the webui. I always got: `404 page not found`

```
And this Log: 
2025-02-07 23:46:18
 23:46:18 rails.1              | I, [2025-02-07T23:46:18.293078 #456]  INFO -- : [098592c1-107f-4b49-8822-bc6f8b956909] {"method":"HEAD","path":"/","format":"*/*","controller":"HomeController","action":"index","status":0,"allocations":380,"duration":3.61,"view":0.0,"db":1.51,"@timestamp":"2025-02-07T23:46:18.292Z","@version":"1","message":"[0] HEAD / (HomeController#index)"}
2025-02-07 23:46:20
 23:46:20 rails.1              | I, [2025-02-07T23:46:20.123572 #456]  INFO -- : [c77c0df1-f7e8-4b08-8d3c-794ccfa09446] {"method":"HEAD","path":"/","format":"*/*","controller":"HomeController","action":"index","status":0,"allocations":379,"duration":1.26,"view":0.0,"db":0.35,"@timestamp":"2025-02-07T23:46:20.123Z","@version":"1","message":"[0] HEAD / (HomeController#index)"}
2025-02-07 23:46:23
 23:46:23 rails.1              | I, [2025-02-07T23:46:23.083124 #456]  INFO -- : [f40dca1e-0af6-464b-aa47-b6a23604bb00] {"method":"HEAD","path":"/","format":"*/*","controller":"HomeController","action":"index","status":0,"allocations":379,"duration":4.78,"view":0.0,"db":0.33,"@timestamp":"2025-02-07T23:46:23.083Z","@version":"1","message":"[0] HEAD / (HomeController#index)"}
```

I removed the Cosmos Proxy URL and mapped the port manually to `3214:3214`

And now its working fine now, but I dont think thats the right way? 

Thanks a lot.

**Comment by waschinski** [comment]
The database URL issue will be fixed once #180 is merged.

I had no issues using port 3214 for the URL except for it showing the URL as not working (red dot) when it is.

**Comment by madejackson** [comment]
as https://github.com/azukaar/cosmos-servapps-official/pull/180 got merged I'm closing this issue. If there is still an issue with this app, please open a new Issue.
